### ISSUE TITLE
Importer cannot load materials before images

### ISSUE THREAD

**Issue by itgenie98** [godotengine/godot#28203]
**Godot version:**
3.1 stable

**OS/device including version:**
Windows 10, GTX 1080, 419.67

**Issue description:**
Imported a bunch of 3D Objects (Wavefront) and Textures at once. Godot couldn't load the Materials (.mtl) properly.

**Steps to reproduce:**
- Create a new Project
- Close Godot
- Make inside the Project a folder and put there some Wavefront Object with its Material Files
- Put also the Textures inside the folder
- Open godot and let it import everything
- Try to place one of the Objects: no texture or material on it
- Console also reports errors like:
```
ERROR: Couldn't open MTL file 'curve_01.mtl', it may not exist or not be readable.
   At: editor/import/resource_importer_obj.cpp:49
WARNING: _parse_material_library: OBJ: Ambient light for material 'street_01' is ignored in PBR
     At: editor/import/resource_importer_obj.cpp:67
ERROR: No loader found for resource: res://Models/Streets/street_base_01.png
   At: core/io/resource_loader.cpp:285
WARNING: _parse_material_library: OBJ: Ambient light for material 'street_01_NONE' is ignored in PBR
     At: editor/import/resource_importer_obj.cpp:67
ERROR: No loader found for resource: res://Models/Streets/street_base_01.png
   At: core/io/resource_loader.cpp:285
```

**Comment by ZainlessBrombie** [comment]
bump still exists and it's a core feature

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

**Comment by itgenie98** [comment]
Yes, it is still reproduceable in 3.2.3.

The Zip contains an Godot project that shows the issue, the bug also survives restarts. the only fix for now is to manually re-import all Models.
I also added the blender source file that I used to create the Wavefront Object.

[ModelImportBug.zip](https://github.com/godotengine/godot/files/5291866/ModelImportBug.zip)

**Comment by KoBeWi** [comment]
Confirmed in 64d3827
The model appears untextured when you remove `.import` folder, but it's fine when you remove only the model file imports.

**Comment by akien-mga** [comment]
Confirmed as still reproducible in recent Godot versions by @and-rad in https://github.com/godotengine/godot/issues/36173#issuecomment-1525699419.

**Comment by and-rad** [comment]
Just to be clear, this is only reproducible on the `3.x` branch. On `4.0.2-stable` and onwards this works without issues.

**Comment by ManPython** [comment]
> Just to be clear, this is only reproducible on the `3.x` branch. On `4.0.2-stable` and onwards this works without issues.

Not true. Problem exist with: https://github.com/godotengine/godot/releases/download/4.2.1-stable/Godot_v4.2.1-stable_win64.exe.zip
